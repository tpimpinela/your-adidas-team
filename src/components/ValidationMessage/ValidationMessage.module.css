.validation {
  width: 100%;
  text-align: center;

  & .validation__header {
    color: var(--accent-color);
    font-size: 1.5rem;
  }

  & .validation__message {
    white-space: break-spaces;
  }
}
