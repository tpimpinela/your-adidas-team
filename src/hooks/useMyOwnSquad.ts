import { useCallback, useContext } from "react";
import { MyOwnSquadContext } from "../contexts/MyOwnSquad.context";

const useMyOwnSquad = () => {
  const myOwnSquad = useContext(MyOwnSquadContext);

  const isMemberOnMySquad = useCallback(
    (teamMemberId: number) => {
      return Boolean(myOwnSquad.find(({ id }) => id === teamMemberId));
    },
    [myOwnSquad]
  );

  return {
    squad: myOwnSquad,
    isMemberOnMySquad,
  };
};

export default useMyOwnSquad;
