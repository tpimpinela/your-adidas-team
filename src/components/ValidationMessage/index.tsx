import useMyOwnSquad from "../../hooks/useMyOwnSquad";
import styles from "./ValidationMessage.module.css";

const ValidationMessage = () => {
  const { validationsMessage } = useMyOwnSquad();

  return (
    <div className={styles.validation}>
      <h3 className={styles["validation__header"]}>
        {validationsMessage
          ? "Your team will be ready when:"
          : "Your team is ready"}
      </h3>
      {validationsMessage && (
        <span
          data-testid="validation-message"
          className={styles["validation__message"]}
        >
          {validationsMessage}
        </span>
      )}
    </div>
  );
};

export default ValidationMessage;
