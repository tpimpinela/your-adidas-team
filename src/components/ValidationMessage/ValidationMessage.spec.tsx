import { render, RenderResult } from "@testing-library/react";
import { it, beforeEach, describe, Mock, vi, expect } from "vitest";
import ValidationMessage from ".";
import useMyOwnSquad from "../../hooks/useMyOwnSquad";

vi.mock("../../hooks/useMyOwnSquad", () => ({
  default: vi.fn(),
}));

describe("<ValidationMessage />", () => {
  let renderResult: RenderResult;
  const useMyOwnSquadMock = useMyOwnSquad as Mock<any, any>;

  describe("there are validation errors", () => {
    const validationMessageMock = "test";

    beforeEach(() => {
      useMyOwnSquadMock.mockReturnValue({
        validationsMessage: validationMessageMock,
      });
      renderResult = render(<ValidationMessage />);
    });

    it("should match the snapshot", () => {
      expect(renderResult.container).toMatchSnapshot();
    });

    it("should show validation message", () => {
      expect(renderResult.getByTestId("validation-message")).toBeTruthy();
      expect(renderResult.getByTestId("validation-message").textContent).toBe(
        validationMessageMock
      );
    });

    it("should show the correct header", () => {
      expect(
        renderResult
          .getByRole("heading", { level: 3 })
          .textContent?.toLowerCase()
      ).toBe("your team will be ready when:");
    });
  });

  describe("there are no validation errors", () => {
    beforeEach(() => {
      useMyOwnSquadMock.mockReturnValue({
        validationsMessage: undefined,
      });
      renderResult = render(<ValidationMessage />);
    });

    it("should match the snapshot", () => {
      expect(renderResult.container).toMatchSnapshot();
    });

    it("should not show validation message", () => {
      expect(renderResult.queryByTestId("validation-message")).toBeFalsy();
    });

    it("should show the correct header", () => {
      expect(
        renderResult
          .getByRole("heading", { level: 3 })
          .textContent?.toLowerCase()
      ).toBe("your team is ready");
    });
  });
});
