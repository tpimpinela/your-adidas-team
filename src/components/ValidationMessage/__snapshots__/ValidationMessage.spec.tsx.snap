// Vitest Snapshot v1

exports[`<ValidationMessage /> > there are no validation errors > should match the snapshot 1`] = `
<div>
  <div
    class="_validation_34c1ba"
  >
    <h3
      class="_validation__header_34c1ba"
    >
      Your team is ready
    </h3>
  </div>
</div>
`;

exports[`<ValidationMessage /> > there are validation errors > should match the snapshot 1`] = `
<div>
  <div
    class="_validation_34c1ba"
  >
    <h3
      class="_validation__header_34c1ba"
    >
      Your team will be ready when:
    </h3>
    <span
      class="_validation__message_34c1ba"
      data-testid="validation-message"
    >
      test
    </span>
  </div>
</div>
`;
